pallets/moonbeam-lazy-migrations/src/lib.rs

```diff
@@ -65,6 +86,209 @@ pub mod pallet {
 		AddressesLengthCannotBeZero,
 		/// The contract is not corrupted (Still exist or properly suicided)
 		ContractNotCorrupted,
+		/// The key lengths exceeds the maximum allowed
+		KeyTooLong,
+	}
+
+	const MAX_ITEM_PROOF_SIZE: u64 = 30 * 1024; // 30 KB
+	const PROOF_SIZE_BUFFER: u64 = 100 * 1024; // 100 KB
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_idle(_n: BlockNumberFor<T>, remaining_weight: Weight) -> Weight {
+			let proof_size_before: u64 = get_proof_size().unwrap_or(0);
+			let res = Pallet::<T>::handle_migration(remaining_weight);
+			let proof_size_after: u64 = get_proof_size().unwrap_or(0);
+			let proof_size_diff = proof_size_after.saturating_sub(proof_size_before);
+
+			Weight::from_parts(0, proof_size_diff)
+				// For now the DbWeight is only recording the ref_time and not account for
```

Comment:
It's not juts "for now", it's the expected behavior and will always be the case

Reply:
Here it mention it will be updated in the future.
https://github.com/paritytech/polkadot-sdk/blob/09254eb9f8670f942c68914ff6fc0ffb503aaf23/substrate/primitives/weights/src/lib.rs#L52-L55

Reply:
This comment make no sense, it's impossible to define the same proof size amount for all reads, it depend on the size of the specific storage item to read

Reply:
You are correct, I will remove this comment for now.